### PR TITLE
Miao Liao-final project 'Working Hours'

### PR DESCRIPTION
**Pull request by Miao-Liao**
This project consists in the replicaion and alternative representation of graphs from Our World in Data called "Working Hours"

### CODE REVIEW

File: _projects/2024/100508856/100508856.Rmd [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+title: "The evolution of working hours in the world"
+description: |
+  This project consists in the replicaion and alternative representation of graphs from Our World in Data called "Working Hours"
+categories: "2023"
+author: Miao Liao
+date: "`r Sys.Date()`"
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: TRUE
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: `true`, in lower case.

File: _projects/2024/100508856/100508856.Rmd [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+preview: original.png
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: This is to show a image as preview in the gallery, but you didn't add a `original.png` file to the pull request. Please, remove this.

File: _projects/2024/100508856/100508856.Rmd [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: Should be `echo=TRUE`. We want the reader to see your code.

File: _projects/2024/100508856/100508856.Rmd [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+```
+
+## Introduction
+
+The graph for this project was inspired by a Working Hours article in Our World in Data, which is  about a comparative analysis of average annual hours worked by countries between 1950 and 2019. The first part of the article discusses the average annual working hours of citizens in different countries during this period, depicted through line graphs and maps. We can clearly observe the trends in average annual working hours for each country, identifying which countries have the longest and shortest average annual working hours. Additionally, the variation in colors on the map represents changes in annual working hours. The second part utilizes data on the annual holidays in different countries during this period to depict trends in annual holidays. The third part includes data on population numbers and GDP of various countries, creating different charts to observe the relationships between population numbers, national GDP, and annual working hours in different countries.You can find the full article [here](Working Hours - Our World in Data).
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: The last link should have the URL to the article between parenthesis.

File: _projects/2024/100508856/100508856.Rmd [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+
+# Figure 1
+library(ggplot2)
+library(dplyr)
+library(ggrepel)
+library(tidyr)
+devtools::install_github("malcolmbarrett/ggokabeito")
+library(ggokabeito)
+library(paletteer)
+working_hours<-read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual working hours/annual-working-hours-per-worker.csv")
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: The CSV is missing in the pull request. Please add this file, and then these lines should be just `read.csv("annual-working-hours-per-worker.csv")`.

File: _projects/2024/100508856/100508856.Rmd [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+                  color = "Region") +
+     scale_x_continuous(breaks = seq(1950, 2017, by = 10)) 
+ 
+p
+```
+
+```{r}
+#Figure 2
+
+data <- read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual working hours/annual-working-hours-per-worker.csv")
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: The data was read before, so you can just reuse the `working_hours` variable.

File: _projects/2024/100508856/100508856.Rmd [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+#Figure 3 maps
+
+library(tmap)
+data("World")
+
+##The map in 1900
+data_1900 <- data[data$Year==1900,] %>% select("Entity","time")
+colnames(data_1900) <- c("sovereignt","Hours")
+world_1900 <- left_join(World,data_1900,by="sovereignt")
+tm_shape(world_1900) +
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: The course is about ggplot2, and we have studied how to plot maps with ggplot2. ;-) Please, use ggplot2 for these maps.

File: _projects/2024/100508856/100508856.Rmd [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+## Part two
+
+## evolution path
+
+The original graph is the evolutionary path of the data of different countries superimposed, and then dynamic simulation is carried out. In this part, I select several regions to draw one of them. Here is a link to the original graph:(https://ourworldindata.org/grapher/annual-working-hours-vs-gdp-per-capita-pwt?time=earliest..2019&country=BRA~USA~DEU~TWN%C2%AEion~PRT~ESP~AUS~NZL~ZAF)
+
+
+```{r}
+
+data <- read.csv( "C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/GDP.csv")
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: This CSV should be added to the pull request too.

File: _projects/2024/100508856/100508856.Rmd [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+
+1. Read file data
+2. Extract and modify column names
+3. Clean data: Remove blank data
+4. Extract the country I want to draw
+5. Use the log10 function to draw the bubble, and use the ggplot2 function to set the x axis, y axis, color, title
+
+
+```{r}
+data <- read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual-working-hours-vs-gdp-per-capita-pwt/annual-working-hours-vs-gdp-per-capita-pwt.csv")
```
Comment (Enchufa2) [inline _projects/2024/100508856/100508856.Rmd]: This is different than before, right? Well... all CSVs should be added to the pull request. ;-)

**Review by Enchufa2 — CHANGES_REQUESTED** [review]
Thanks for your work! Please address the comments below and let me know when this is ready for review again.

**Comment by Miao-Liao** [comment]
Thank you so much ! I've modified the misstakes.

**Comment by Enchufa2** [comment]
The CSVs are still missing in the PR. I meant that they should be move inside your project directory, not to C:\. In this way, they will appear in the git panel, and then you can commit and push them here.

**Comment by Miao-Liao** [comment]
Thank you and I've changed the question about read. csv, I hope it will be OK

**Review by Enchufa2 — APPROVED** [review]
Perfect, thanks! Mergin with some fixes.

**Comment by Enchufa2** [comment]
Here's your work: https://csslab.uc3m.es/dataviz/projects/2023/100508856/

**Pull request merged**